gui-js/apps/minsky-electron/src/app/managers/CommandsManager.ts

```diff
@@ -58,23 +58,43 @@ export class CommandsManager {
       : classTypeRes;
 
     if (!classType) {
-      return;
+      return undefined;
     }
     return ClassType[classType];
   }
 
-  static async getItemInfo(x: number, y: number): Promise<CanvasItem> {
+  static isFalseResult(result) {
+    return !result || typeof result === 'object' && (Object.keys(result).length === 0);
+  }
 
-    if (!minsky.canvas.getItemAt(x, y)) {
+  static async getItemInfo(x: number, y: number): Promise<CanvasItem> {
+    if (this.isFalseResult(await minsky.canvas.getItemAt(x, y))) {
       return null;
     }
 
+    return await this.getCurrentItemInfo();
+  }
+
+  static async getFocusItemInfo(): Promise<CanvasItem> {
+    await minsky.canvas.setItemFromItemFocus();
```

Comment:
I still think await is valuable here because it communicates that the actions are executed sequentially, even through that is obvious to someone more versed on the technical workings of the backend command queue.